alternator/server.cc

```diff
@@ -208,7 +208,8 @@ class local_nodelist_handler : public gated_handler {
         // just the list of live nodes in this DC needs more elaborate code:
         auto& topology = _proxy.get_token_metadata_ptr()->get_topology();
         sstring local_dc = topology.get_datacenter();
-        std::unordered_set<gms::inet_address> local_dc_nodes = topology.get_datacenter_endpoints().at(local_dc);
+        auto dc_endpoints_map = topology.get_datacenter_endpoints();
+        const auto& local_dc_nodes = dc_endpoints_map.at(local_dc);
         for (auto& ip : local_dc_nodes) {
             if (_gossiper.is_alive(ip)) {
                 rjson::push_back(results, rjson::from_string(ip.to_sstring()));
```

Comment:
I guess since we return a ready future below, the lack of capture isn't an active bug, just one waiting to happen.

Reply:
yes.  we should use a get_token_metadata() method instead to make it clear we're returning a reference to the token_metadata du jour.